whylog/teacher/__init__.py

```diff
@@ -27,12 +83,16 @@ def update_pattern(self, line_id, proposed_pattern):
     def make_group(self, line_id, span):
```

Comment:
I don't understand what this function do. I though that user is adding new group by writing new regex pattern. The docString is misleading. I think you spend more time writing docString, then the time you'll write this function. For me it block the projekt.